packages/beacon-node/src/chain/produceBlock/produceBlockBody.ts

```diff
@@ -135,38 +140,9 @@ export async function produceBlockBody<T extends BlockType>(
   //     logger.error("Attestation did not transfer to op pool", {}, e);
   //   }
   // }
-
-  const stepsMetrics =
-    blockType === BlockType.Full
-      ? this.metrics?.executionBlockProductionTimeSteps
-      : this.metrics?.builderBlockProductionTimeSteps;
-
-  const [attesterSlashings, proposerSlashings, voluntaryExits, blsToExecutionChanges] =
-    this.opPool.getSlashingsAndExits(currentState, blockType, this.metrics);
-
-  const endAttestations = stepsMetrics?.startTimer();
-  const attestations = this.aggregatedAttestationPool.getAttestationsForBlock(this.forkChoice, currentState);
-  endAttestations?.({
-    step: BlockProductionStep.attestations,
-  });
-
-  const endEth1DataAndDeposits = stepsMetrics?.startTimer();
-  const {eth1Data, deposits} = await this.eth1.getEth1DataAndDeposits(currentState);
-  endEth1DataAndDeposits?.({
-    step: BlockProductionStep.eth1DataAndDeposits,
-  });
-
-  const blockBody: phase0.BeaconBlockBody = {
-    randaoReveal,
-    graffiti,
-    eth1Data,
-    proposerSlashings,
-    attesterSlashings,
-    attestations,
-    deposits,
-    voluntaryExits,
-  };
-
+  const blockBody = phase0BlockBody ?? (await produceBlockBodyPhase0.call(this, blockType, currentState, blockAttr));
+  const blsToExecutionChanges = this.opPool.getBlsToExecutionChanges(currentState, blockType, this.metrics);
```

Comment:
move this also into produceCommonBlockBody, so whatever which isn't dependent on execution payload/header, which is almost all fields for now since all other fields are independent